### PR TITLE
Add first-class Azure login support

### PR DESCRIPTION
**Pull request by benashz**

### CODE REVIEW

File: internal/provider/auth_azure.go [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+		credOpts.ID = azidentity.ClientID(v.(string))
+	}
+
+	creds, err := azidentity.NewManagedIdentityCredential(credOpts)
+	if err != nil {
+		return "", err
+	}
+
+	tOpts := policy.TokenRequestOptions{
+		Scopes: []string{"https://management.azure.com/"},
```
Comment (vinay-gopalan) [inline internal/provider/auth_azure.go:177]:
```suggestion
		Scopes: []string{defaultAzureScope},
```

File: internal/provider/auth.go [hunk 2]
```diff
@@ -80,19 +81,57 @@ func (l *AuthLoginCommon) Method() string {
...
 		delete(params, k)
 	}
 
 	return params
 }
 
+func (l *AuthLoginCommon) copyParamsIncluding(includes ...string) (map[string]interface{}, error) {
+	missing := []string{}
+	params := map[string]interface{}{}
+	for _, k := range includes {
```
Comment (vinay-gopalan) [inline internal/provider/auth.go:120]: nit: some duplicated code in here and `copyParams`; wondering if it's worth refactoring

File: internal/provider/auth.go [hunk 3]
```diff
@@ -80,19 +81,57 @@ func (l *AuthLoginCommon) Method() string {
...
+func (l *AuthLoginCommon) copyParamsExcluding(excludes ...string) map[string]interface{} {
+	params, _ := l.copyParams()
 	for _, k := range excludes {
 		delete(params, k)
 	}
 
 	return params
 }
 
+func (l *AuthLoginCommon) copyParamsIncluding(includes ...string) (map[string]interface{}, error) {
```
Comment (vinay-gopalan) [inline internal/provider/auth.go:117]: I don't see this being used currently. Is this something we'll use for later?

**Review by vinay-gopalan — APPROVED** [review]
Looking good! A couple comments/questions

File: internal/provider/auth.go (hunk 3 above)
Comment (benashz) [reply internal/provider/auth.go]: Fixed in 2855a9dc194b4b3cf2caf0935b8997265b78f979

File: internal/provider/auth.go (hunk 2 above)
Comment (benashz) [reply internal/provider/auth.go]: Fixed in 2855a9dc194b4b3cf2caf0935b8997265b78f979

File: internal/provider/auth_azure.go (hunk 1 above)
Comment (benashz) [reply internal/provider/auth_azure.go]: Fixed in 2855a9dc194b4b3cf2caf0935b8997265b78f979

Comment (vinay-gopalan) [reply internal/provider/auth_azure.go]: Ah nice!

**Pull request merged**